app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewModel.kt

```diff
@@ -198,37 +209,73 @@ class ConversationViewModel @Inject constructor(
         }
     }
 
+    fun downloadAsset(messageId: String) {
+        viewModelScope.launch {
+            if (!isAssetDownloaded(messageId))
+                updateAssetMessageDownloadStatus(IN_PROGRESS, conversationId, messageId)
+
+            val result = getRawAssetData(conversationId, messageId)
+            updateAssetMessageDownloadStatus(if (result != null) DOWNLOADED else FAILED, conversationId, messageId)
+
+            if (result != null) {
+                val assetName = getAssetName(messageId)
+                showOnAssetDownloadedDialog(assetName, result)
+            }
+        }
```

Comment:
do we want to proceed with this logic regardless if the asset is downloaded or not ? because as for now this piece of code will be executed regardless if isAssetDownloaded is true or false

Reply:
This logic is indeed needed to either download the asset or fetch it from the local database. After that, we will show the dialog to ask the user what to do with the file.
I renamed the method to `downloadOrFetchAsset` and added some documentation to make it clearer.